pkg/operator/controller/ingress/controller.go

```diff
@@ -1131,6 +1131,10 @@ func (r *reconciler) ensureIngressController(ci *operatorv1.IngressController, d
 
 	SetIngressControllerNLBMetric(ci)
 
+	if err := r.updateInfraIngressLoadBalancerIPs(lbService, platformStatus); err != nil {
+		errs = append(errs, fmt.Errorf("failed to Ingress Load Balancer IPs to Infrastructure PlatformStatus: %v", err))
+	}
+
```

Comment:
`lbService` could be nil, and `updateInfraIngressLoadBalancerIPs` doesn't have a nil check.  You need to check `haveLB` here or add a nil check in `updateInfraIngressLoadBalancerIPs`.  I prefer the former because we prefer to use Booleans instead of nil checks in this operator, and we happen to have a Boolean here.
```suggestion
	if haveLB {
		if err := r.updateInfraIngressLoadBalancerIPs(lbService, platformStatus); err != nil {
			errs = append(errs, fmt.Errorf("failed to Ingress Load Balancer IPs to Infrastructure PlatformStatus: %v", err))
		}
	}
```